plugins/org.python.pydev.core/pysrc/_pydev_bundle/pydev_import_hook.py

```diff
@@ -14,25 +18,23 @@ def add_module_name(self, module_name, activate_function):
         self._modules_to_patch[module_name] = activate_function
 
     def do_import(self, name, *args, **kwargs):
-        activate_func = None
-        if name in self._modules_to_patch:
-            activate_func = self._modules_to_patch.pop(name)
-
         module = self._system_import(name, *args, **kwargs)
+
         try:
+            activate_func = self._modules_to_patch.get(name)
             if activate_func:
-                activate_func() #call activate function
+                activate_func()  # call activate function
+                del self._modules_to_patch[name]
         except:
-            sys.stderr.write("Matplotlib support failed\n")
```

Comment:
`stderr.write` here causes issues as the error hits the user's console for what is an internal error.  Is there somewhere better this can be logged to?